cluster-install/install.sh

```diff
@@ -2450,13 +2533,19 @@ step_generate_and_apply_resource_limit_patches() {
     # itself does "support" resource limits, in the sense that they don't get cleared out when we set them.
 
     # Trident Operator
-    if ! trident_will_be_installed_or_modified; then
+    if [ -z "$_USER_ACCEPTED_TRIDENT_OPERATOR_UPGRADE" ]; then
+        logdebug "no trident upgrade operator was needed, skipping trident operator resource limits"
+    elif [ "$_USER_ACCEPTED_TRIDENT_OPERATOR_UPGRADE" != "true" ]; then
+        logdebug "user declined trident operator upgrade, skipping trident operator resource limits"
+    elif ! trident_will_be_installed_or_modified; then
         logdebug "skipping trident operator resource limits"
     elif create_kubectl_patch_for_containers "deploy/trident-operator" "$trident_ns" "$patch_path" "$patch_value"; then
         patches_list_for_debugging+=("$_return_value")
         apply_kubectl_patch "$_return_value"
+    elif [ -n "$_return_error" ]; then
+        add_problem "Failed to create resource limit patch for the Trident Operator deployment: $_return_error"
     else
-        add_problem "Failed to create resource limit patch for the Trident Operator deployment (unknown error)"
+        add_problem "Failed to create resource limit patch for the Trident Operator deployment: unknown error"
```

Comment:
As mentioned above, the `create_kubectl_patch_for_containers` function was calling `fatal` on some errors that were instead more appropriate as "problems" to be communicated to the user properly. These changes are to handle those errors.